### DISCUSSION TITLE
Problem with Merging Metadata and Tracking Data from Tracab

### DISCUSSION THREAD

**Discussion by impact-data** [PySport/kloppy#352] [Q&A] [answered]
I encountered an issue with the code indicating that the <match> element is missing in the metadata file. I downloaded the data from Tracab, but unfortunately, the element seems to be absent from the Tracab metadata. Here is the code I used:

from kloppy import tracab

dataset = tracab.load(
    meta_data="../../kloppy/tests/files/tracab_meta.xml",
    raw_data="../../kloppy/tests/files/tracab_raw.dat",
    
    # Optional arguments
    sample_rate=1/2,
    limit=10,
    coordinates="tracab",
    only_alive=True
)

# Because this is a test dataset the output contains some garbage
dataset.to_df().head()

Below is a screenshot of the metadata file for your reference:

![screen_metadata](https://github.com/user-attachments/assets/8ddefc58-dc0b-4a9d-965b-ec297cb61cb8)

![element](https://github.com/user-attachments/assets/594800a6-a0f9-4aec-b48a-1ab30305237a)

**Comment by UnravelSports** [comment]
I had a quick look, but is there any chance you can share the meta data file as an XML with as little modification as possible? (It's fine if you want to anonymize it etc).

Because looking at the meta_data parser we're looking for the `period` and `iStartFrame` and `iEndFrame` tags to identify when the period started, but from having a quick look at your screenshot it seems as though these tags don't exist and have been replaced with something else (namely `Phase1StartFrame` etc.). This might indicate Tracab created a new version of their meta data file.

So, then it would be very helpful if you could share the actual XML so we can build a parser for the new version of the meta data too.

@JanVanHaaren @koenvossen @probberechts thoughts?

**Comment by impact-data** [comment]
Thank you for your prompt response. I have included the metadata attachment in the zip file. @UnravelSports 

[fixtures.zip](https://github.com/user-attachments/files/17154586/fixtures.zip)

**Comment by UnravelSports** [comment, marked as answer]
@impact-data I've created a PR to fix this here: https://github.com/PySport/kloppy/pull/353
